lib/DxilPIXPasses/DxilShaderAccessTracking.cpp

```diff
@@ -586,38 +592,39 @@ DxilShaderAccessTracking::GetResourceFromHandle(Value *resHandle,
       auto properties = hlsl::resource_helper::loadPropsFromAnnotateHandle(
           annotateHandle, *DM.GetShaderModel());
 
-      auto* handleCreation = cast<CallInst>(annotateHandle.get_res());
-
-      if (hlsl::OP::IsDxilOpFuncCallInst(handleCreation, hlsl::OP::OpCode::CreateHandleFromBinding)) {
-          DxilInst_CreateHandleFromBinding createHandleFromBinding(handleCreation);
-          Constant* B = cast<Constant>(createHandleFromBinding.get_bind());
-          auto binding = hlsl::resource_helper::loadBindingFromConstant(*B);
-          ret.accessStyle = AccessStyle::FromRootSig;
-          ret.index = createHandleFromBinding.get_index();
-          ret.registerType = RegisterTypeFromResourceClass(
-              static_cast<hlsl::DXIL::ResourceClass>(binding.resourceClass));
-          ret.RegisterSpace = binding.spaceID;
-      } else if (hlsl::OP::IsDxilOpFuncCallInst(handleCreation, hlsl::OP::OpCode::CreateHandleFromHeap)) {
-          DxilInst_CreateHandleFromHeap createHandleFromHeap(handleCreation);
-          ret.accessStyle = createHandleFromHeap.get_samplerHeap_val()
-              ? AccessStyle::SamplerFromDescriptorHeap : AccessStyle::ResourceFromDescriptorHeap;
-          ret.dynamicallyBoundIndex = createHandleFromHeap.get_index();
-
-          ret.registerType = RegisterTypeFromResourceClass(properties.getResourceClass());
-
-          DynamicResourceBinding drb{};
-          drb.HeapIsSampler = createHandleFromHeap.get_samplerHeap_val();
-          drb.HeapIndex = -1;
-          drb.Name = "ShaderNameTodo";
-          if (auto * constInt = dyn_cast<ConstantInt>(createHandleFromHeap.get_index()))
-          {
-              drb.HeapIndex = constInt->getLimitedValue();
-          }
-          m_dynamicResourceBindings.emplace_back(std::move(drb));
-
-          return ret;
-      } else {
-          DXASSERT_NOMSG(false);
+      auto* handleCreation = dyn_cast<CallInst>(annotateHandle.get_res());
+      if (handleCreation != nullptr) {
```

Comment:
The diff in this block is smaller than github makes it look. The only change is cast->dyn_cast on 595, the if on 596, and the extra closing brace on 627